backends/p4tools/modules/testgen/targets/bmv2/backend/ptf/ptf.cpp

```diff
@@ -260,11 +248,18 @@ class AbstractTest(bt.P4RuntimeTest):
         pass
 
     def runTestImpl(self):
-        self.setupCtrlPlane()
-        logger.info("Sending Packet ...")
-        self.sendPacket()
-        logger.info("Verifying Packet ...")
-        self.verifyPackets()
+        # TODO: This is a workaround for broken unit tests on Ubuntu 20.04. It can happen that,
+        # when the last test fails, the failure output of that test is swallowed by the unit test
```

Comment:
what is the unit test framework?
where is the bug?

Reply:
I will remove this. The bug is in this PR https://github.com/p4lang/p4c/pull/3875 but this approach does not solve it. I can not reproduce these failures on Ubuntu 20.04 or 22.04. It looks like some CI problem. The unit test framework is Python `unittest`, which PTF relies on.